core/services/gateway/delegate.go

```diff
@@ -54,7 +56,12 @@ func (d *Delegate) ServicesForSpec(ctx context.Context, spec job.Job) (services
 	if err2 != nil {
 		return nil, errors.Wrap(err2, "unmarshal gateway config")
 	}
-	handlerFactory := NewHandlerFactory(d.legacyChains, d.ds, d.lggr)
+	// TODO: pass in TLS config and default HTTP timeout
+	httpClient, err := network.NewHttpClient(nil, time.Duration(10)*time.Millisecond, d.lggr)
```

Comment:
Some networking configs such as max response size and write timeout seem like global node settings. One other potential advantage is it can have its own state and allow additional logic in the future (e.g. keep track of endpoint health to determine wait time before retrying)

Reply:
OK, fair.